src/fmeval/eval_algorithms/context_quality.py

```diff
@@ -0,0 +1,291 @@
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed context quality scores.
+
+        :param record: the input record.
+        :return: The input record with the context quality scores added.
+        """
+        model_input = record[self.model_input_key]
+        target_output = record[self.target_output_key]
+        retrieved_context = record[self.retrieved_context_key]
+        record[self.context_precision_score_key] = self._get_score(model_input, target_output, retrieved_context)
+        return record
+
+    def _get_score(self, model_input: str, target_output: str, retrieved_context: str) -> float:
+        """Computes the context precision score by composing a prompt on each (context chunk, model input,
+        target output) triple and doing inference on the judge model to get a verdict of 0 or 1, then
+        calculating the score using the judge model verdicts.
+
+        For multiple target contexts, we do a "logical or" operation on the model verdicts, so if the
+        model verdict for a given context chunk is 1 for any target output, the model verdict for that
+        context chunk will be 1.
+
+        :param model_input: The input that is composed with a prompt template and passed to the model.
+        :param target_output: The referenced "ground truth" answer.
+        :param retrieved_context: The context retrieved from the RAG system.
+        :return: the context precision score.
+        """
+        target_outputs = target_output.split(self.target_output_delimiter)
+        context_chunks = retrieved_context.split(self.retrieved_context_delimiter)
+        results = []
+        for context_chunk in context_chunks:
+            model_verdict = 0
+            for target_output in target_outputs:
+                prompt = self.context_precision_prompt_composer.compose(
+                    model_input, {"target_output": target_output, "retrieved_context": context_chunk}
+                )
+                model_output = self.judge_model.predict(prompt)[0]
+                model_verdict = ContextPrecisionScore._parse_model_output(model_output)
+                if model_verdict:
+                    break
+            results.append(model_verdict)
+        score = ContextPrecisionScore._compute_metric(results)
+        return score
+
+    @staticmethod
+    def _parse_model_output(model_output: str) -> int:
+        """Parses the model output to a binary score of 0 or 1
+
+        :param model_output: The output of the judge model.
+        :return: Score of 0 or 1.
+        """
+        response_words = model_output.split(" ")
+        predicted_labels = [
+            word.lower().strip() for word in response_words if word.lower().strip() in BINARY_SCORE_VALUES
+        ]
+        string_label = predicted_labels[0] if predicted_labels else "0"
+        return int(string_label)
```

Comment:
Since we're effectively looking for the string `"1"`, can we simplify this function to just

```
return int("1" in response_words)
```

Reply:
Right. I guess my question is why we always take the first one. For example, if we have `predicted_labels = ["0", "1", ...]`, why should we return `"0"` instead of `"1"`? It seems quite arbitrary at the moment.

Reply:
This is mostly following what we do for [classification accuracy](https://github.com/aws/fmeval/blob/1b356df244e00d563d3715392b6aaa35f314b47e/src/fmeval/eval_algorithms/classification_accuracy.py#L76) for converting the model output. Will see if science has any suggestions on this.